src/nvim/generators/gen_api_dispatch.lua

```diff
@@ -4,43 +4,65 @@ mpack = require('mpack')
 if arg[1] == '--help' then
   print('Usage: genmsgpack.lua args')
   print('Args: 1: source directory')
-  print('      2: dispatch output file (dispatch_wrappers.generated.h)')
-  print('      3: functions metadata output file (funcs_metadata.generated.h)')
-  print('      4: API metadata output file (api_metadata.mpack)')
-  print('      5: lua C bindings output file (msgpack_lua_c_bindings.generated.c)')
+  print('      2: dispatch output file (dispatch_wrappers.generated.c)')
+  print('      3: dispatch table output file (dispatch_table.generated.h)')
+  print('      4: functions metadata output file (funcs_metadata.generated.h)')
+  print('      5: API metadata output file (api_metadata.mpack)')
+  print('      6: lua C bindings output file (msgpack_lua_c_bindings.generated.c)')
   print('      rest: C files where API functions are defined')
 end
-assert(#arg >= 4)
+assert(#arg >= 6)
 functions = {}
 
 local nvimdir = arg[1]
-package.path = nvimdir .. '/?.lua;' .. package.path
+package.path = nvimdir .. '/?/init.lua;' .. nvimdir .. '/?.lua;' .. package.path
+package.path = nvimdir .. '/../../?.lua;' .. package.path
 
 -- names of all headers relative to the source root (for inclusion in the
 -- generated file)
 headers = {}
+-- Like `headers`, but without headers ending with .generated.h
+local written_headers = {}
 
--- output h file with generated dispatch functions
-dispatch_outputf = arg[2]
+-- output c file with generated dispatch functions
+local dispatch_outputf = arg[2]
+-- output h file with generated dispatch table
+local dispatch_table_outputf = arg[3]
 -- output h file with packed metadata
-funcs_metadata_outputf = arg[3]
+local funcs_metadata_outputf = arg[4]
 -- output metadata mpack file, for use by other build scripts
-mpack_outputf = arg[4]
-lua_c_bindings_outputf = arg[5]
+local mpack_outputf = arg[5]
+-- output c file with lua bindings
+local lua_c_bindings_outputf = arg[6]
 
 -- set of function names, used to detect duplicates
 function_names = {}
 
 c_grammar = require('generators.c_grammar')
+local lust = require('generators.lust')
+local gperf = require('generators.gperf')
+local global_test_helpers = require('test.helpers')
 
--- read each input file, parse and append to the api metadata
-for i = 6, #arg do
-  local full_path = arg[i]
+local dedent = global_test_helpers.dedent
+local shallowcopy = global_test_helpers.shallowcopy
```

Comment:
@justinmk It is what I was asking in some comment: should it be moved and where if yes.